AP_HAL_ChibiOS: add new target Stellar H7V2 (#28926)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/StellarH7V2/Readme.md

## Features
    Processor
        STM32H743VIH6 480 MHz, 2MB flash
    Sensors
        ICM-42688p Acc/Gyro with external clock feature
        DPS310/BMP280 barometer
        AT7456E OSD
        SD Card
    Power
        2S-8S Lipo input voltage with voltage monitoring
        12V, 3A BEC for powering Video Transmitter
        5V, 2A BEC for internal and peripherals
    Interfaces
        10x PWM outputs DShot capable, 4 outputs BDShot capable
        7x UARTs
        1x CAN
        1x I2C
        3x ADC
        SD card for logging
        USB-C port
    LED
        Red, 3.3V power indicator
        Blue and Green, FC status
    Size
        41 x 41mm PCB with 30.5mm M3 mounting

  
## Overview

![StellarH7V2](Stellar_H7V2.png)

![StellarH7V2](StellarH7V2.png)

## UART Mapping

The UARTs are marked Rx* and Tx* in the above pinouts. The Rx* pin is the
receive pin for UART*. The Tx* pin is the transmit pin for UART*. All UARTS are DMA capable.

 - SERIAL0 -> USB
 - SERIAL1 -> UART1 (MAVLink2)
 - SERIAL2 -> UART2 (MAVLink2)
 - SERIAL3 -> UART3 (User)
 - SERIAL4 -> UART4 (Serial RC input)
 - SERIAL5 -> UART6 (GPS)
 - SERIAL6 -> UART7 (DisplayPort)

>  - SERIAL7 -> UART8 (ESC Telemetry)

Reviewer: ```suggestion
 - SERIAL7 -> UART8 (ESC Telemetry, RX8 pin only)
```

## CAN and I2C

StellarH7V2 supports 1x CAN bus and 1x I2C bus

## RC Input

The default RC input is configured on the UART4 RX4 input and can be used for all ArduPilot supported unidirectional receiver protocols. 
* PPM is not supported.
* SBUS/DSM/SRXL connects to the RX4 pin.

> * FPort requires connection to TX4 and RX4 via a bi-directional inverter. See :ref:`common-FPort-receivers`.

Reviewer: ```suggestion
* FPort requires connection to TX4  and :ref:`SERIAL4_OPTIONS<SERIAL4_OPTIONS>` set to "7". See :ref:`common-FPort-receivers`.
```

> * CRSF also requires a TX4 connection, in addition to RX6, and automatically provides telemetry.

Reviewer: ```suggestion
* CRSF also requires a TX4 connection, in addition to RX4, and automatically provides telemetry.
```

* SRXL2 requires a connection to TX4 and automatically provides telemetry. Set :ref:`SERIAL4_OPTIONS<SERIAL4_OPTIONS>` to “4”.

   
## OSD Support

StellarH7V2 supports using its internal OSD using OSD_TYPE 1 (MAX7456 driver). Simulatenous DisplayPort OSD operation  is preconfigured on SERIAL 6 but requires OSD_TYPE2 = 5. See :ref:`common-msp-osd-overview-4.2` for more info.

## PWM Output

StellarH7V2 supports up to 10 PWM outputs. PWM 1-4 outputs support DShot.

All the channels support DShot. Channels 1-4 support bi-directional DShot. PWM outputs are grouped and every group must use the same output protocol:
* 1, 2, 3, 4 are Group 1;
* 5, 6, 7, 8 are Group 2;

> * 9, 10      are Group 3;

Reviewer: the above groups are incorrect

Reviewer: where is M9 and M10 pinned out...not on board photo

Reviewer: and Andy's request to reformat the bidir lines has not been made

## Battery Monitoring

The board has 1 built-in voltage dividers and 2x current ADC. support external 3.3V based current sensor

The voltage input is compatible with 2~8S LiPo batteries.
The default battery parameters are:
* :ref:`BATT_MONITOR<BATT_MONITOR>` = 4
* :ref:`BATT_VOLT_PIN<BATT_VOLT_PIN__AP_BattMonitor_Analog>` = 10

> * :ref:`BATT_CURR_PIN<BATT_CURR_PIN__AP_BattMonitor_Analog>` = 11

Reviewer: ```suggestion
* :ref:`BATT_CURR_PIN<BATT_CURR_PIN__AP_BattMonitor_Analog>` = 11 (CURR pin)
```

* :ref:`BATT_VOLT_MULT<BATT_VOLT_MULT__AP_BattMonitor_Analog>` = 11
* :ref:`BATT_AMP_PERVLT<BATT_AMP_PERVLT__AP_BattMonitor_Analog>` = 10

> * :ref:`BATT2_CURR_PIN<BATT2_CURR_PIN__AP_BattMonitor_Analog>` = 7

Reviewer: ```suggestion
* :ref:`BATT2_CURR_PIN<BATT2_CURR_PIN__AP_BattMonitor_Analog>` = 7 (ADC2 pin)
```

> # StellarH7V2 Flight Controller

Reviewer 2: Needs to be two commits, one for each subsystem. The commit message should start with:
"bootloaders: " - for the bootloader changes and
"AP_HAL_ChibiOS: " - for the hwdef changes

# StellarH7V2 Flight Controller

https://stingbee.com.ua/flight_controllers/stellarh7v2

## Features
    Processor
        STM32H743VIH6 480 MHz, 2MB flash
    Sensors
        ICM-42688p Acc/Gyro with external clock feature
        DPS310/BMP280 barometer
        AT7456E OSD
        SD Card
        W25N02G/1G dataflash
    Power
        8S Lipo input voltage with voltage monitoring
        12V, 3A BEC for powering Video Transmitter
        5V, 2A BEC for internal and peripherals
    Interfaces

>         10x PWM outputs DShot capable

Author: yes

>         10x PWM outputs DShot capable

Reviewer: where is PWM8 and 9?

        7x UARTs
        1x CAN
        1x I2C
        3x ADC
        SD card and 128/256MB NAND for logging
        USB-C port
    LED
        Red, 3.3V power indicator
        Blue and Green, FC status
    Size
        41 x 41mm PCB with 30.5mm M3 mounting

  
## Overview

![StellarH7V2](Stellar_H7V2.png)

![StellarH7V2](StellarH7V2.png)

## Wiring Diagram

## UART Mapping

The UARTs are marked Rx* and Tx* in the above pinouts. The Rx* pin is the
receive pin for UART*. The Tx* pin is the transmit pin for UART*.

 - SERIAL0 -> USB

>  - SERIAL1 -> USART1 (User)

Reviewer 2: These need some better descriptions I think

 - SERIAL2 -> USART2 (User)
 - SERIAL3 -> USART3 (User)
 - SERIAL4 -> UART4  (Serial RC input)
 - SERIAL6 -> USART6 (User)
 - SERIAL7 -> USART6 (User)
 - SERIAL8 -> USART6 (User)

## CAN and I2C

StellarH7V2 supports 1x CAN bus and 1x I2C bus
multiple CAN peripherals can be connected to one CAN bus in parallel. similarly for I2C bus.

## RC Input

RC input is configured on the UART4(SERIAL4). It supports all serial RC protocols. SERIAL2_PROTOCOL=23 by default.

   
## OSD Support

> StellarH7V2 supports using its internal OSD using OSD_TYPE 1 (MAX7456 driver). External OSD support such as DJI or DisplayPort is supported using any spare UART. See :ref:`common-msp-osd-overview-4.2` for more info.

Reviewer: ```suggestion
StellarH7V2 supports using its internal OSD using OSD_TYPE 1 (MAX7456 driver). Simulatenous DisplayPort OSD operation  is preconfigured on SERIAL 6 but requires OSD_TYPE2 = 5. See :ref:`common-msp-osd-overview-4.2` for more info.
```

## PWM Output

StellarH7V2 supports up to 10 PWM outputs. All outputs support DShot.

> 

Reviewer: ```suggestion
All the channels support DShot. Channels 1-4 support bi-directional DShot. PWM outputs are grouped and every group must use the same output protocol:

* 1, 2,are Group 1;
* 3 - 6 are Group 2;
* 7 - 9 are Group 3;
```

## Battery Monitoring

The board has 1 built-in voltage dividers and 2x current ADC. support external 3.3V based current sensor

> 

Reviewer: ```suggestion
The voltage input is compatible with 2~8S LiPo batteries.

The default battery parameters are:

* :ref:`BATT_MONITOR<BATT_MONITOR>` = 4
* :ref:`BATT_VOLT_PIN<BATT_VOLT_PIN__AP_BattMonitor_Analog>` = 10
* :ref:`BATT_CURR_PIN<BATT_CURR_PIN__AP_BattMonitor_Analog>` = 11
* :ref:`BATT_VOLT_MULT<BATT_VOLT_MULT__AP_BattMonitor_Analog>` = 11
* :ref:`BATT_AMP_PERVLT<BATT_AMP_PERVLT__AP_BattMonitor_Analog>` = 10
* :ref:`BATT2_CURR_PIN<BATT2_CURR_PIN__AP_BattMonitor_Analog>` = 7
```

## Compass

StellarH7V2 does not have a built-in compass, but you can attach an external compass using I2C on the SDA and SCL pads.

> 

Reviewer: need to add section on Camera Switch:
```suggestion
## Camera Switch
GPIO 81 controls which camera input (CAM1 or CAM22 is applied to the internal OSD. A RELAY function can be enabled to control the switching.
```
